specs/Pills.md

```diff
@@ -0,0 +1,182 @@
+# [Pills]()
+
+## Background
+
+PIlls should be used when representing an input, as a way to filter content, or to represent an attribute.
+
+## Prior Art
+
+- [Evergreen](https://evergreen.segment.com/components/badge-and-pill/)
+- [Lightning](https://www.lightningdesignsystem.com/components/pills/)
+- [Material UI](https://material-ui.com/components/chips/#chip)
+- [OpenUI Research](https://github.com/WICG/open-ui/pull/259)
+
+## Sample Code
+
+```jsx
+<Pill>Item 1</Pill>
+<Pill clickable>Item 2</Pill>
+<TogglePill>On</TogglePill>
+```
+
+## Variants
+
+- Shape: `circular`, `rounded`
+- Size: `smaller`, `small`, `medium`
+- Appearance: `Filled Grey (filled)`, `Filled White (inverted)`, `Outline`
+
+## PROPS
+
+```typescript
+export type PillShape = 'rounded' | 'circular';
+
+export type PillAppearence = 'filled' | 'inverted' | 'outline';
+
+export type SizeValue = 'smaller' | 'small' | 'medium';
+
+export interface PillProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /**
+   * A Pill can be sized.
+   */
+  size?: SizeValue;
+
+  /**
+   * A Pill can be circular or rounded
+   */
+  shape?: PillShape;
+
+  /**
+   * A Pill can be filled, inverted or outline
+   */
+  appearance?: PillVariant;
+
+  /**
+   * Media slot
+   */
+  media?: ShorthandProps<HTMLElement>;
+
+  /**
+   * Details that will appear as second line
+   */
+  details?: ShorthandProps<HTMLElement>;
+
+  /** A Pill can be dismissible. */
+  dismissible?: boolean;
+
+  /** A Pill can be disabled. */
+  disabled?: boolean;
+
+  /** A Pill can be selected. */
+  selected?: boolean;
+
+  /** A Pill can be clickable */
+  clickable?: boolean;
+
+  /** A callback to be called when Pill is clicked */
+  onClick?: function;
+
+  /**
+   * A button shorthand for the dismiss action slot. To use this slot the pill should be
+   * dismissible.
+   */
+  dismissIcon?: ShorthandProps<HTMLElement>;
```

Comment:
Is `dismissIcon` mandatory for `<Pill dismissible />? If not, is there a default dismiss icon or no icon is rendered?

Reply:
By default no icon is rendered, if the `Pill` is dismissible we render the default `X` icon or what users provides.